client/packages/glsp-sprotty/src/base/command-stack.ts

```diff
@@ -61,6 +75,12 @@ export class ObservableCommandStack extends CommandStack {
         super.handleCommand(command, operation, beforeResolve)
     }
 
+    get model(): Promise<SModelRoot> {
```

Comment:
At second glance this implementation of IModelAccess is problematic. Since readonly !== immutable it's possible to retrieve a model via the IModelAccess interface and modify it outside of a command which can lead to unwanted side effects and inconsistencies. (e.g. changes done via IModelAccess are not visible until another command is executed and a model update is invoked, no undo/redo for this changes etc.)

We can avoid this issue by using the modelfactory (basically creating a deep copy) to avoid this issue.
Changes made to the exposed model will not effect the actual model anymore.
```typescript
get model(): Promise<SModelRoot> {
        return this.currentPromise.then(
            state => this.modelFactory.createRoot(state.root)
        )
}
```

Reply:
Yes, that's a good comment! I addressed that in the changes below.